WCFDataService/Client/System/Data/Services/Client/MaterializeFromAtom.cs

```diff
@@ -69,6 +69,11 @@ internal class MaterializeAtom : IDisposable, IEnumerable, IEnumerator
         private bool moved;
 
         /// <summary>
+        /// The ODataResponseMessage to dispose when we are done.
+        /// </summary>
+        private IODataResponseMessage responseMessage;
```

Comment:
Aparently this will not dispose the message (or the stream)...
            var e = ctx.Execute<File>(new Uri("http://host/OData/Files"));
            var en = e.GetEnumerator();
            en.Dispose();

But this will...
            var e = ctx.Execute<File>(new Uri("http://host/OData/Files"));
            var en = e.GetEnumerator();
            en.MoveNext();
            en.Dispose();

We can fix this scenario.